.circleci/scripts/git-diff-develop.js

```diff
@@ -0,0 +1,76 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+const fs = require('fs');
+const path = require('path');
+
+
+async function fetchWithDepthIncrement(depth) {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error) {
+    console.error(`Failed to fetch with depth ${depth}:`, error.message);
+    return false;
+  }
+}
+
+async function gitDiffWithRetry() {
+  const depths = [1, 10, 100];
+  let diffOutput = '';
+
+  for (const depth of depths) {
+    try {
+      console.log(`Attempting git diff with depth ${depth}...`);
+      await fetchWithDepthIncrement(depth);
+
+      const { stdout: diffResult } = await exec(`git diff --name-only origin/develop...${process.env.CIRCLE_BRANCH}`);
```

Comment:
Nit: In each of these places where we use `origin/develop`, perhaps we could use `origin/HEAD` instead? That should represent the default branch, and will keep working if/when we change it from `develop`.

Reply:
great suggestion! Changed :+1: